File: src/test/java/org/foonugget/kdict/UtilTest.java

package org.foonugget.kdict;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class UtilTest {

    @Test
    public void test() {
        assertEquals("it''s", Util.escapeDatabaseStringLiteral("it's"));
        assertEquals("its''", Util.escapeDatabaseStringLiteral("its'"));
        assertEquals("its", Util.escapeDatabaseStringLiteral("its"));
        assertEquals("", Util.escapeDatabaseStringLiteral(""));
    }

}

File: app/src/test/java/org/foonugget/kdict/UtilTest.kt
package org.foonugget.kdict

import org.junit.Assert.assertEquals
import org.junit.Test

class UtilTest {

    @Test
    fun testEscapeDatabaseStringLiteral() {
        assertEquals("it''s", escapeDatabaseStringLiteral("it's"))
        assertEquals("its''", escapeDatabaseStringLiteral("its'"))
        assertEquals("its", escapeDatabaseStringLiteral("its"))
        assertEquals("", escapeDatabaseStringLiteral(""))
    }

    private fun escapeDatabaseStringLiteral(string: String): String {
        return string.replace("'", "''")
    }
}
